TezosKit/Operation/OperationFactory.swift

```diff
@@ -93,21 +93,44 @@ public class OperationFactory {
   ///   - amount: The amount of XTZ to transact.
   ///   - from: The address that is sending the XTZ.
   ///   - to: The address that is receiving the XTZ.
-  ///   - parameters: Optional parameters to include in the transaction if the call is being made to a smart contract.
   ///   - operationFees: OperationFees for the transaction. If nil, default fees are used.
   public func transactionOperation(
     amount: Tez,
     source: Address,
     destination: Address,
-    parameters: [String: Any]? = nil,
     operationFees: OperationFees? = nil
   ) -> Operation {
     let operationFees = operationFees ?? defaultFeeProvider.fees(for: .transaction, in: tezosProtocol)
     return TransactionOperation(
       amount: amount,
       source: source,
       destination: destination,
-      parameters: parameters,
+      operationFees: operationFees
+    )
+  }
+
+  /// Create a new smart contract invocation operation.
+  ///
+  /// - Parameters:
+  ///   - contract: The smart contract to invoke.
+  ///   - amount: The amount of Tez to transfer with the invocation.
+  ///   - parameter: An optional parameter to send to the smart contract.
+  ///   - source: The address invoking the contract.
+  ///   - signatureProvider: The object which will sign the operation.
+  ///   - operationFees: OperationFees for the transaction. If nil, default fees are used.
+  public func smartContractInvocationOperation(
```

Comment:
Convenience functions for calling contracts with parameters but 0 amount and without parameters or amount might be useful

Reply:
Agree. 

This is an internal class to TezosNodeClient. I'll set a parameter default to zero on the TezosNodeClient, which is the public gateway to this API.